<?xml version="1.0"?>

<launch>

    <arg name="world_frame_" default="map" />
    <arg name="base_frame_" default="map" />
    <arg name="car_start_x_" default= "default" />
    <arg name="car_start_y_" default="default" />
    <arg name="car_start_orientation_" default="default" />
    
    <arg name="vel_topic_" default="default" />

    <!-- World/Map frame name -->
    <param name="world_frame" value="map" />
    
    <!-- Set the maximum velocity here -->
    <param name="/move_base/TebLocalPlannerROS/max_vel_x" value="1.0" />

    <!-- Get the configuration file for the car -->
    <rosparam command="load" file="$(find pomdp_car_control)/config/car.yaml" />
    <param name="world_frame" value="$(arg world_frame_)" />
    <param name="base_frame" value="$(arg base_frame_)" />
    <param name="car_start_x" value="$(arg car_start_x_)" />
    <param name="car_start_y" value="$(arg car_start_y_)" />
    <param name="car_start_orientation" value="$(arg car_start_orientation_)" />
    
   <param name="vel_topic" value="$(arg vel_topic_)" />


    <!-- Launch the Control Node -->
    <node name="car_control" pkg="pomdp_car_control" type="pomdp_car_control_node" respawn="true" output="screen" />

</launch>
